Importação das bibliotecas

In [2]:
import pandas as pd
import numpy as np

Carregando dataset

In [4]:
df = pd.read_csv("../dados/Varejo.csv", sep=';')

Análise exploratória dos dados

In [10]:
print(f'Quantidade de registros encontrados: {df.shape[0]}')
print(f'Quantidade de colunas encontradas: {df.shape[1]}')


Quantidade de registros encontrados: 830000
Quantidade de colunas encontradas: 14


In [12]:
print('\n Início do dataset: \n')
df.head()


 Início do dataset: 



,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_ID,PR_CAT,PR_NOME,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
0,01/02/2019,1000,534,M,4,1,C,67,BEBIDAS,REFRIGERANTE GUARANA,NaN,NaN,NaN,NaN
1,01/02/2019,1000,534,M,4,1,C,70,BEBIDAS,REFRIGERANTE OUTROS,NaN,NaN,NaN,NaN
2,01/02/2019,1000,534,M,4,1,C,178,HIGIENE,LENCO UMEDECIDO,NaN,NaN,NaN,NaN
3,01/02/2019,1000,534,M,4,1,C,4,ALIMENTOS,ABACAXI,NaN,NaN,NaN,NaN
4,01/02/2019,1000,534,M,4,1,C,175,LIMPEZA,LIMPADOR MULTIUSO,NaN,NaN,NaN,NaN


In [ ]:
print('\n Final do dataset: \n')
df.tail()


 Final do dataset: 



,DATA,CO_ID,CL_ID,CL_GENERO,CL_EC,CL_FHL,CL_SEG,PR_ID,PR_CAT,PR_NOME,Unnamed: 10,Unnamed: 11,Unnamed: 12,Unnamed: 13
829995,19/08/2022,919822,155,F,2,0,B,183,ALIMENTOS,KETCHUP,NaN,NaN,NaN,NaN
829996,19/08/2022,919822,155,F,2,0,B,56,ALIMENTOS,QUEIJO MUSSARELA,NaN,NaN,NaN,NaN
829997,19/08/2022,919822,155,F,2,0,B,227,ALIMENTOS,ARROZ,NaN,NaN,NaN,NaN
829998,19/08/2022,919822,155,F,2,0,B,214,ALIMENTOS,CEBOLA,NaN,NaN,NaN,NaN
829999,19/08/2022,919822,155,F,2,0,B,59,ALIMENTOS,SALGADINHO,NaN,NaN,NaN,NaN


In [14]:
print("Tipos de dados")
print(df.dtypes)

Tipos de dados
DATA               str
CO_ID            int64
CL_ID            int64
CL_GENERO          str
CL_EC            int64
CL_FHL           int64
CL_SEG             str
PR_ID            int64
PR_CAT             str
PR_NOME            str
Unnamed: 10    float64
Unnamed: 11    float64
Unnamed: 12    float64
Unnamed: 13    float64
dtype: object


Segunda etapa da análise - Nulos, duplicados e estatísticas gerais

In [15]:
print("Valores nulos por coluna")
print(df.isnull().sum())

Valores nulos por coluna
DATA                0
CO_ID               0
CL_ID               0
CL_GENERO           0
CL_EC               0
CL_FHL              0
CL_SEG              0
PR_ID               0
PR_CAT              0
PR_NOME             0
Unnamed: 10    830000
Unnamed: 11    830000
Unnamed: 12    830000
Unnamed: 13    830000
dtype: int64


In [16]:
print('Duplicatas')
print(f'Quantidade de registros duplicados: {df.duplicated().sum()}')

Duplicatas
Quantidade de registros duplicados: 96553


In [17]:
print('Estatísticas gerais')
print(df.describe())

Estatísticas gerais
              CO_ID          CL_ID          CL_EC        CL_FHL  \
count  830000.00000  830000.000000  830000.000000  830000.00000   
mean   460045.08989     499.595204       2.596911       1.14654   
std    265465.24754     287.568253       1.170599       1.41696   
min      1000.00000       1.000000       1.000000       0.00000   
25%    233117.00000     254.000000       2.000000       0.00000   
50%    456517.00000     498.000000       3.000000       0.00000   
75%    690132.00000     746.000000       4.000000       2.00000   
max    919822.00000    1000.000000       5.000000       4.00000   

               PR_ID  Unnamed: 10  Unnamed: 11  Unnamed: 12  Unnamed: 13  
count  830000.000000          0.0          0.0          0.0          0.0  
mean      115.051855          NaN          NaN          NaN          NaN  
std        66.129241          NaN          NaN          NaN          NaN  
min         1.000000          NaN          NaN          NaN          NaN  
2

Problemas encontrados até aqui:
- Data está em formato str e deveria ser datetime;
- Existem 4 colunas em que todos os dados estão nulos;
- Existem 96553 registros duplicados no dataframe.

Vamos continuar investigando!

Tratando e limpando a base de dados

In [18]:
# Removendo as colunas que estão completamente vazias

colunas_vazias = df.columns[df.isnull().all()]
print(f'Colunas completamente vazias que serão removidas: {list(colunas_vazias)}')

df = df.drop(columns = colunas_vazias)

Colunas completamente vazias que serão removidas: ['Unnamed: 10', 'Unnamed: 11', 'Unnamed: 12', 'Unnamed: 13']


In [ ]:
# Convertendo DATA para datetime

df['DATA'] = pd.to_datetime(
    df['DATA'],
    format="%d/%m/%Y",
    errors='coerce'
)

data_invalida = df['DATA'].isna().sum()

print(f'\nDatas inválidas encontradas: {data_invalida}')
print(f'\nTipo de dado da data: {df['DATA'].dtypes}')


Datas inválidas encontradas: 0

Tipo de dado da data: datetime64[us]
